enterprise/kernel/src/main/java/org/neo4j/kernel/impl/store/format/highlimit/NodeRecordFormat.java

```diff
@@ -114,8 +113,8 @@ protected byte headerBits( NodeRecord record )
     @Override
     protected boolean canUseFixedReferences( NodeRecord record )
     {
-        return (((record.getNextProp() == NULL) || ((record.getNextProp() & THREE_BITS_OVERFLOW_BIT_MASK) == 0)) &&
-                ((record.getNextRel() == NULL) || ((record.getNextRel() & ONE_BIT_OVERFLOW_BIT_MASK) == 0)));
+        return (((record.getNextProp() != NULL) && ((record.getNextProp() & HIGH_DWORD_LOWER_NIBBLE_MASK) == 0)) &&
+                ((record.getNextRel() != NULL) && ((record.getNextRel() & HIGH_DWORD_LOWER_NIBBLE_MASK) == 0)));
```

Comment:
The read method with its `BaseRecordFormat.longFromIntAndMod` calls should also handle them.